src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java

```diff
@@ -104,12 +105,22 @@ public static ExperimentTypeFetchOptions fetchExperimentTypesCompletely() {
     return experimentTypeFetchOptions;
   }
 
-  public static List<PropertyType> getPropertiesOfEntityType(IEntityType type) {
-  List<PropertyType> res = new ArrayList<>();
-  List<PropertyAssignment> assignments = type.getPropertyAssignments();
-  for (PropertyAssignment propertyAssignment : assignments) {
-    res.add(propertyAssignment.getPropertyType());
+  public static List<PropertyType> getPropertiesOfEntityType(List<IEntityType> foundTypes) {
+
+    if (foundTypes.isEmpty()) {
+      throw new NotFetchedException("Entity type could not be found");
+    }
+    if (foundTypes.size() > 1) {
+      throw new NotFetchedException("More than one entity type found");
+    }
```

Comment:
Then why do you want a list if you only expect to have one element?

Reply:
because that's what is returned by the search function of openbis - these are basically checks.